contracts/TokenFactory.sol

```diff
@@ -335,28 +321,63 @@ contract TokenFactory is ReentrancyGuard, Ownable {
 
         address winnerToken = getWinnerByCompetitionId(_competitionId);
 
-        require(winnerToken != tokenAddress, "token address is the winner");
-
         Token token = Token(tokenAddress);
-        uint256 burnedAmount = token.balanceOf(msg.sender);
-
-        uint256 receivedETH = _sell(
-            tokenAddress,
-            burnedAmount,
-            msg.sender,
-            address(this)
-        );
 
-        uint256 mintedAmount = _buy(winnerToken, msg.sender, receivedETH);
-
-        emit BurnTokenAndMintWinner(
-            msg.sender,
-            tokenAddress,
-            winnerToken,
-            burnedAmount,
-            receivedETH,
-            mintedAmount,
-            block.timestamp
-        );
+        if(winnerToken == tokenAddress) {
+            address pool = _createLiquilityPool(tokenAddress);
+
+            tokensPools[tokenAddress] = pool;
+
+            uint256 ethAmount = collateralById[_competitionId][tokenAddress];
+
+            address creator = tokensCreators[tokenAddress];
+            uint256 creatorBalance = token.balanceOf(creator);
+
+            // burned all creator tokens    
+            token.burn(creator, creatorBalance);
```

Comment:
Token creator's tokens shouldn't be burned and seized. Otherwise there is no incentive for them to create tokens and market that